src/server/AppManager.ts

```diff
@@ -1032,6 +1041,30 @@ export class AppManager {
 
         return result;
     }
+
+    private checksum(str: string, alg = 'SHA256'): string {
+        return createHash(alg).update(str).digest('hex');
+    }
+
+    private stringifyOrdered(obj: any): string {
+        return JSON.stringify(obj, Object.keys(obj).sort());
+    }
+
+    private async getAppSignature(obj: any, appId: string): Promise<string> {
+        const descriptorString = this.stringifyOrdered(obj);
+        const calg = 'SHA256';
+        const alg = 'RS512';
+        const checksum = this.checksum(descriptorString, calg);
+        const pkeyString = await AppManager.Instance.getBridges().getInternalFederationBridge().getPrivateKey();
```

Comment:
Curious to why you needed the `AppManager.Instance` part if you are inside the `AppManager` class 🤔